inc/classes/rest-api/class-media-library.php

```diff
@@ -1396,7 +1402,26 @@ public function get_media_folders( $request ) {
 			$terms = $this->get_all_children_terms( $terms, $taxonomy );
 		}
 
-		return rest_ensure_response( $this->prepare_term_responses( $terms ) );
+		// Prepare the response data.
+		$prepared_terms = $this->prepare_term_responses( $terms );
+
+		// Ensure we always have an array, never null.
+		if ( is_null( $prepared_terms ) || ! is_array( $prepared_terms ) ) {
+			$prepared_terms = array();
+		}
+
+		// Create the response.
+		$response = rest_ensure_response( $prepared_terms );
+
+		// Add headers only for paginated requests.
+		if ( ! $locked && ! $bookmark ) {
+			$response->header( 'X-WP-Total', $total_items );
+			$response->header( 'X-WP-TotalPages', $total_pages );
+			$response->header( 'X-WP-Current-Page', $page );
+			$response->header( 'X-WP-Per-Page', $per_page );
```

Comment:
```suggestion
			$response->header( 'X-Wp-Total', $total_items );
			$response->header( 'X-Wp-Totalpages', $total_pages );
			$response->header( 'X-Wp-Current-Page', $page );
			$response->header( 'X-Wp-Per-Page', $per_page );
```

Because, response is converting it to these headers name, So we can make this consistent.

<img width="1439" height="679" alt="Image" src="https://github.com/user-attachments/assets/34ede830-918e-447d-80e8-0d9aed1cdaef" />